include/mp/proxy-io.h

```diff
@@ -247,7 +247,7 @@ struct Waiter
     template <typename Fn>
     void post(Fn&& fn)
     {
-        std::unique_lock<std::mutex> lock(m_mutex);
+        std::unique_lock<std::mutex> lock(m_mutex); // NOLINT(misc-const-correctness)
```

Comment:
In commit "clang-tidy: Fix `misc-const-correctness` check" (06806fe90ce1f565f19432d13bc7635dd7e6a9f4)

Maybe just declare these const? It seems more stable than adding NOLINT. Especially because if any an unlock call is made to any of these locks later the lint errors would go away but nolint comments would still be there.

Reply:
re: https://github.com/chaincodelabs/libmultiprocess/pull/144#discussion_r1939769026

This is done in #152